### PR TITLE
Add tests for 4 screens

### PR DESCRIPTION
**Pull request by andrew-pogue**
- `PackagesTest::testPackagesUI`
- `PickupMethodTest::testPickupMethodUI`
- `PickupDateTest::testPickupDateUI`
- `PricingTest::testPricingUI`
- improved `composetools/PickupMethod.kt`

### CODE REVIEW

**Comment by andrew-pogue** [comment]
all tests pass except `LiveChatTest::sendMessage` and `LiveChatTest::sendMessageChatInput`

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.example.returnpals.PackageInfo
+import com.example.returnpals.PackageLabelType
+import org.junit.Rule
+import org.junit.Test
+
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:13]: Should we comment on our code here and in the test functions?

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            )
+        }
+
+        // Test next and back buttons:
+
+        val next = rule.onNodeWithText("Next")
+        val back = rule.onNodeWithText("Back")
+
+        next.assertExists("Next button does not exist")
+        back.assertExists("Back button does not exist")
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:40]: I like how thorough you are with this.

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        assert(isClicked) { "Next button not enabled when packages table is non-empty." }
+
+        packages[2] = PackageInfo(2, "jcpenny.png", PackageLabelType.PHYSICAL)
+        rule.onNodeWithText("jcpenny.png").assertIsDisplayed()
+
+        val physical = rule.onNodeWithText("Physical Label")
+        val digital = rule.onNodeWithText("Digital Label")
+        val qrcode = rule.onNodeWithText("Amazon QR Code")
+
+        physical.assertExists("Add physical label button does not exist.")
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:67]: With all these asserts in one function, if the first assert fails how would we know if the rest of the UI is working?  Could any of this be broken into smaller components for testing and troubleshooting?

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PickupDateTest.kt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            PickupDateUI(
+                date = date.value,
+                onChangeDate = { date.value = it },
+                onClickBack = { isClicked = true },
+                onClickNext = { isClicked = true },
+                isValidDate = { Log.println(Log.INFO, "date", it.toString()); it >= minDate && it < maxDate }
+            )
+        }
+
+        // Test next and back buttons:
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/PickupDateTest.kt:35]: Same thought as the other file. Could this be separated into different functions for testing and troubleshooting?

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/ProfileTest.kt [hunk 5]
```diff
@@ -19,7 +21,7 @@ class ProfileTest {
     @Test
     fun planTest(){
         rule.setContent{ ProfileContent() }
-        rule.onNodeWithText("SILVER").assertIsDisplayed()
+        rule.onNodeWithTag(PricingPlan.SILVER.toString()).assertIsDisplayed()
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/ProfileTest.kt:24]: Thanks for fixing this.

File: app/src/main/java/com/example/returnpals/ScheduleReturn.kt [hunk 6]
```diff
@@ -29,130 +34,161 @@ import java.time.LocalDate
...
-        _pickup.update {
-            it.packages.remove(id)
-            idManager.free(id)
-            it
-        }
+        _data.value!!.packages.remove(id)
+        _idManager.free(id)
+    }
+
+    fun onChangePlan(value: PricingPlan) {
```
Comment (emptyloot) [inline app/src/main/java/com/example/returnpals/ScheduleReturn.kt:118]: These are looking great. I like the naming makes everything clear. Should we still add our function comment?

**Review by emptyloot — CHANGES_REQUESTED** [review]
Could you add a comment at the top of the new test functions that is a summary of what the function does? You have comments in the function that say what it does, could you also add them to the top?

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt (hunk 3 above)
Comment (andrew-pogue) [reply app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:67]: I can break it up into multiple parts. However, we would have multiple tests for one function like `testPricingUIBackButton`, `testPricingUINextButton`, `testPricingUIBronzeButton`, .... which might be OK.

Comment (andrew-pogue) [reply app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:67]: I will make a commit to break it up

File: app/src/main/java/com/example/returnpals/ScheduleReturn.kt (hunk 6 above)
Comment (andrew-pogue) [reply app/src/main/java/com/example/returnpals/ScheduleReturn.kt:118]: Schedule Return ViewModel is still in development, it'll be implemented, commented, and tested a by the end of this sprint.

File: app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+/**
+ * Tests the [PackagesUI] composable.
+ */
+class PackagesTest {
+
+    @get:Rule
+    val rule = createComposeRule()
+
+    /**
+     * Test the next button. Check that it is displayed, that the on-click callback works, and that it can be disabled.
```
Comment (emptyloot) [inline app/src/androidTest/kotlin/com/example/returnpals/composetools/PackagesTest.kt:24]: Thanks for adding this to follow the code policy

File: app/src/main/java/com/example/returnpals/ScheduleReturn.kt
Comment (emptyloot) [inline app/src/main/java/com/example/returnpals/ScheduleReturn.kt:1]: I would like to put it onto an issue for the end of the quarter to add comments to older code for policy. If you refactor again we could add it then too.

File: app/src/main/java/com/example/returnpals/composetools/PickupMethod.kt [hunk 8]
```diff
@@ -67,171 +68,150 @@ fun PickupMethodUI(
...
+                )
+            },
+        shape = RoundedCornerShape(22.dp, 22.dp, 22.dp, 22.dp),
         content = content
     )
 }
 
 @Composable
-private fun HandOffDescription(
+private fun HandOffTMethodContent(
```
Comment (emptyloot) [inline app/src/main/java/com/example/returnpals/composetools/PickupMethod.kt:148]: Thanks for renaming these.

**Review by emptyloot — APPROVED** [review]
Thank you for adding comments. We can continue to work on fixing our older code when we are refactoring.

**Pull request merged**